tests/services/githubevent.service.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { GithubEventService } from '../../src/app/core/services/githubevent.service';
+import { of } from 'rxjs';
+import { EVENTS, ADD_LABEL_EVENT, CHANGE_TITLE_EVENT } from '../constants/githubevent.constants';
+
+let githubService: any;
+let issueService: any;
+
+describe('GithubEventService', () => {
+  beforeAll(() => {
+    githubService = jasmine.createSpyObj('GithubService', ['fetchEventsForRepo']);
+  });
+
+  describe('.setLatestChangeEvent()', () => {
+    it('stores the time of the most recent issue event\ and most recent issue update.', async () => {
+      githubService.fetchEventsForRepo.and.returnValue(of(EVENTS));
+      const githubEventService: GithubEventService = new GithubEventService(githubService, null);
```

Comment:
Is it possible to have this in beforeAll() so that we don't need initialise it in each test case?

Reply:
We cannot use `beforeAll` (since the instances are specific to each test case).
We can use `beforeEach` so that the `githubEventService` instance is created before each test case.
However, I feel that the current approach is more readable.

Reason: In our current approach, we instantiate the 2 dependencies of the `GithubEventService` (`IssueService` and `GithubService`) **before** instantiating the `GithubEventService`.

If we create the `GithubEventService` in a `beforeEach` function, we would still have to modify its dependencies afterwards. To me, this seems trickier to understand.

Reply:
I see, okay. That is a fair point. 👍